src/Microsoft.AspNetCore.WebUtilities/FileBufferingReadStream.cs

```diff
@@ -135,22 +180,27 @@ private Stream CreateTempFile()
         public override int Read(byte[] buffer, int offset, int count)
         {
             ThrowIfDisposed();
-            if (_buffer.Position < _buffer.Length || _completelyBuffered)
+            if (_buffer.Position < _readIntoBuffer || _completelyBuffered)
```

Comment:
Hmm... poked around in MemoryStream, looks like that would work; had some issues when trying to get it to work before but maybe I was changing capacity